enhancements/service-catalog/service-catalog-and-broker-removal.md

```diff
@@ -87,69 +84,103 @@ cluster-svcat-controller-manager-operator, with a [Job](https://kubernetes.io/do
 This might be accomplished with just one of the operators, that would need
 a little more investigation.
 
-Replace the operator's code with a new Job executable that will perform the
-deletion of the apiserver associated with the Service Catalog. The following items
-would be removed:
-
-- ClusterServiceClasses
-- ClusterServicePlans
-- ClusterServiceBrokers
-- ServiceClasses
-- ServicePlans
-- ServiceBrokers
-- ServiceInstances
-- ServiceBindings
-
 The idea is that the Job will run during the upgrade until completion. Once
-finished, the Service Catalog will have been removed from the cluster.
-
-The Service Catalog operators would *remain* in the release payload but the
+finished, the Service Catalog Operators will have been removed from the
+cluster. The Job that will delete the Service Catalog operators and their
+associated items.
+
+- cluster-svcat-apiserver-operator
+  - Role named `prometheus-k8s in `openshift-service-catalog-apiserver-operator`
+  - RoleBinding named `prometheus-k8s in `openshift-service-catalog-apiserver-operator`
+  - ServiceMonitor in `openshift-service-catalog-apiserver-operator`
+  - Namespace `openshift-service-catalog-apiserver-operator` (not sure
+    if we can do this from the Job)
+  - ServiceCatalogAPIServer (cr)
+  - ConfigMap named `openshift-service-catalog-apiserver-operator-config`
+  - Service named `metrics`
+  - ClusterRoleBinding for `openshift-service-catalog-apiserver-operator` ServiceAccount
+  - ServiceAccount `openshift-service-catalog-apiserver-operator`
+  - Deployment named `openshift-service-catalog-apiserver-operator`
+  - ClusterOperator named `service-catalog-apiserver`
+- cluster-svcat-controller-manager-operator
+  - Role named `prometheus-k8s in `openshift-service-catalog-controller-manager-operator`
+  - RoleBinding named `prometheus-k8s in `openshift-service-catalog-controller-manager-operator`
+  - ServiceMonitor in `openshift-service-catalog-controller-manager-operator`
+  - Namespace `openshift-service-catalog-controller-manager-operator` (not sure
+    if we can do this from the Job)
```

Comment:
Non-blocking question:

Are you thinking that the Job would be deployed in the namespace that it needs to delete, and there might be a conflict with that? Do we know if CVO will try to recreate the Job if it gets deleted (i.e. since the Job will be deleted if it is in a namespace that gets deleted). Can the Job be created in a separate namespace?

Reply:
That's exactly what my concern is. I'm not entirely sure yet :)